### ISSUE TITLE
Add-on Store Repositories stop getting loaded

### ISSUE THREAD

**Issue by richard-scott** [home-assistant/supervisor#3404]
### Describe the issue you are experiencing

Since the supervised store was merged into the Configuration menu I have not had reliable usage of the Add-on Store, and I no longer get updates for add-ons already installed. When you look at the list of Managed add-on repos, you do not see ones you have added before (and have installed from before such as the Official Add-ons repo:

![](https://user-images.githubusercontent.com/5149834/149495122-1d138864-98f6-41df-ac69-d0c46b68b936.png)

As you can see in this screen shot above, I have things like Mosquitto installed that is no longer available for updates.
To make this worse, when you try and add the repo back in you get told that you can't as its already there:

![](https://user-images.githubusercontent.com/5149834/149495338-0ecf2ee9-a588-40a4-903c-438085453300.png)

### What is the used version of the Supervisor?

supervisor-2021.12.2

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian

### What is the version of your installed operating system?

11.2

### What version of Home Assistant Core is installed?

core-2021.12.9

### Steps to reproduce the issue

Add extra repos and wait.
At some point the icons will vanish and go back to being green puzzle bits.

### Anything in the Supervisor logs that might be useful for us?

```txt
22-01-14 10:32:32 INFO (MainThread) [supervisor.resolution.fixups.store_execute_remove] Remove invalid Store: core
22-01-14 10:32:32 ERROR (MainThread) [supervisor.store.repository] Can't remove built-in repositories!
22-01-14 10:32:32 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
```

**Comment by richard-scott** [comment]
I've added a Github Personal Access Token to my /root/.git-credentials file and it seems a bit more stable now.
Maybe this could be added in the menu in the same/similar way that HACS sets one up.

**Comment by mdegat01** [comment]
I'm confused how this happened but you're trying to readd the wrong repository. You're putting in the link to the community addon's repository in your screenshot which are already listed, hence why it's telling you it's a duplicate. The repository you're missing is the official addon's repository, the url for that is [https://github.com/home-assistant/addons](https://github.com/home-assistant/addons).

I don't really know how you ended up missing the official addon's repository in your list. That shouldn't happen so any additional info on how you got to that state would be appreciated. But at least I think using the correct repo link will get you back up and running again.

**Comment by richard-scott** [comment]
I **_was_** adding back in that community repo URL as my store didn't show any content for it. I then get told I already have it. As you can see in the error message, I used the correct URL.

When my network gets rate limited at Github I lose my repos. That us until I added a PAT to my /root/.git-credentials file on my installation and now going on 3 days now and my repos are still working fine.

**Comment by richard-scott** [comment]
The first picture I sent before (with VerneMQ in the Local Add-ons) was to show you that my Official add-ons repo wasn't working correctly as the icons had reverted back to green puzzle parts.

Basically, I think my issue all comes down to being rate limited by Github and having a nicer way to add a Personal Access Token would be a great leap forward.

**Comment by mdegat01** [comment]
Interesting ok, is there any logs you can share showing the rate limiting error? Should be something in the supervisor log I would think. 

Also you said you put the PAT in `/root/.git-credentials`, can you clarify in which container you mean? Core, Supervisor and each addon runs in its own container and the `/root` folder isn't shared between any of them. Also `/root` isn't preserved over updates in any of the containers so I'm not sure your fix is a permanent one, would help to know more.

**Comment by richard-scott** [comment]
I put that .git-credentials file into the root user of the host running Docker.
I'll have a look for any relevant logs and report back.

**Comment by richard-scott** [comment]
ok, so it seems to be linked to DNS (which is working fine from the docker host running supervisor).

```
22-02-01 08:52:01 INFO (MainThread) [supervisor.store.git] Cloning add-on https://github.com/home-assistant/addons repository
22-02-01 08:52:01 ERROR (MainThread) [supervisor.store.git] Can't clone https://github.com/home-assistant/addons repository: Cmd('git') failed due to: exit code(128)
  cmdline: git clone -v --recursive --depth=1 --shallow-submodules https://github.com/home-assistant/addons /data/addons/core
  stderr: 'Cloning into '/data/addons/core'...
fatal: unable to access 'https://github.com/home-assistant/addons/': Could not resolve host: github.com
'.
22-02-01 08:52:01 INFO (MainThread) [supervisor.resolution.fixups.store_execute_remove] Remove invalid Store: core
22-02-01 08:52:01 ERROR (MainThread) [supervisor.store.repository] Can't remove built-in repositories!
22-02-01 08:52:01 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
```

The VirtualBox VM that is running supervised home assistant is running on the same physical hardware that is my local DNS server. DNS is not in a VM. So one can't be online without the other being operational.

Maybe it could be changed that the repo isn't removed on the first sniff of a DNS delay?

**Comment by mdegat01** [comment]
Hm so two things to check:

1. If you go [here](https://my.home-assistant.io/redirect/supervisor_logs/) and change "Log Provider" from "Supervisor" to "DNS", is there anything notable in the logs related to this?
2. You mentioned having a local DNS server, do you block access to `1.1.1.1` and/or `1.0.0.1` on your network by any chance? My understanding is cloudflare's DNS servers are used as a fallback by supervisor's DNS Plugin but it can get into an issue where it thinks the local DNS server is down and goes straight to the fallback. There's a number of issues related to it [here](https://github.com/home-assistant/plugin-dns/issues).

If you fall into #2 you might have to unblock cloudflare's DNS service specifically for the device running HA for now until those issues can get resolved.

**Comment by richard-scott** [comment]
The DNS issues seem to have fixed themselves so I think we can ignore that. The major fix for me seems to have been adding the .git-credentials file.
